lib/ansible/modules/windows/win_dns_client.ps1

```diff
@@ -43,6 +41,26 @@ Function Write-DebugLog {
     }
 }
 
+Function ConvertTo-AnsibleCompatibleObject {
+    <#
+        .SYNOPSIS
+        Adds a ContainsKey method to any object so Get-AnsibleParam can be used
+        to easily get and verify properties from it.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(ValueFromPipeline=$true)]
+        [Object]$Obj
+    )
+
+    Process {
+        $Obj | Add-Member -MemberType ScriptMethod -Name ContainsKey -Value {
+            param($key)
+            $this.PSObject.Properties.Name.Contains($key)
+        } -Force -PassThru
+    }
+}
```

Comment:
@jborean93 curious what you think about this. After our discussion on checking for properties I realized `Get-AnsibleParam` fits the bill nicely, and its description seems to do exactly what I needed. Only issue was that it requires `.ContainsKey()` on the object, even though the rest of the logic works perfectly with both objects and `[hashtable]`s, so it's helpful for these situations operating in strict mode.

If you think this is more widely useful I'd be open to moving it into `Ansible.ModuleUtils.Legacy`, or updating the logic in `Get-AnsibleParam` to just work with both objects and dictionaries.